likecoin/admin/views/matters.php

```diff
@@ -94,13 +94,73 @@ function likecoin_append_footer_link_element( $dom_document ) {
 	$dom_document->documentElement->appendChild( $p );
 }
 
+/**
+ * Parse and modify post HTML to replace Matters asset url and div/class standard
+ *
+ * @param string|  $post_id raw post HTML content.
+ * @param WP_Post| $post Post object.
+ */
+function likecoin_upload_url_image_to_matters( $post_id, $post ) {
+	if ( ! $post ) {
+		return;
+	}
+	$content = $post->post_content;
+	if ( ! $content ) {
+		return $content;
+	}
+	$dom_document          = new DOMDocument();
+	$libxml_previous_state = libxml_use_internal_errors( true );
+	$dom_content           = $dom_document->loadHTML( '<template>' . mb_convert_encoding( $content, 'HTML-ENTITIES' ) . '</template>', LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD );
+	libxml_clear_errors();
+	libxml_use_internal_errors( $libxml_previous_state );
+	if ( false === $dom_content ) {
+		return $content;
+	}
+	$images             = $dom_document->getElementsByTagName( 'img' );
+	$current_image_urls = array();
+	foreach ( $images as $image ) {
+		$image_infos          = get_post_meta( $post_id, LC_MATTERS_IMAGE_INFO, true );
+		$url                  = $image->getAttribute( 'src' );
+		$current_image_urls[] = $url;
+		$image_url            = $url;
+		// check if $image_url already existed in the post.
+		if ( ! empty( $image_infos ) ) {
+			if ( isset( $image_infos->$image_url ) ) { // if existed in matters, don't need to upload to matters.
+				$image_info = $image_infos->$image_url;
+				if ( is_array( $image_info ) || is_object( $image_info ) ) {
+					if ( $image_info->original_url === $image_url ) {
+						continue;
+					}
+				}
+			}
+		}
+		likecoin_post_url_image_to_matters( $image_url, $image_infos ); // not in matters, need to upload to matters.
+		$image_infos = get_post_meta( $post_id, LC_MATTERS_IMAGE_INFO, true );
```

Comment:
Why do we need to get_post_meta() here? Can we take the return value of `likecoin_post_url_image_to_matters` and just insert it into `$image_infos` without an extra call to `get_post_meta()`?

Reply:
Yes :)